src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -535,6 +661,20 @@ xclLoadAxlf(const axlf *buffer)
   if (force_program) {
     flags = flags | DRM_ZOCL_FORCE_PROGRAM;
   }
+
+#if defined(XRT_ENABLE_LIBDFX)
+  // check OVERLAY section
+  // if present use libdfx apis to load bitstream and dtbo(overlay)
+  auto overlay_header = xclbin::get_axlf_section(buffer, axlf_section_kind::OVERLAY);
+  if(overlay_header) {
+    ret = libdfxLoadAxlf(this->mCoreDevice, buffer, overlay_header, mKernelFD, flags, dtbo_id);
```

Comment:
I looked through code and I don't see where this `mKernelFD` is closed?
Using raw fd that are opened and manually closed is very poor design.  RAII should be used.

Reply:
Hi @stsoe, mKernelFD is member of shim class, it is closed in destructor of shim.